Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 5c181cccc90110dae82c0045d1a853170898a6b6 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Mon, 25 Feb 2019 15:00:26 +0200
Subject: [PATCH 27/42] [ozone/wayland] Return primary display on null window.

WaylandScreen::GetDisplayForAcceleratedWidget might be
called right after the browser closed the WaylandWindow (on
shutting down, for example), which may result in segfault or DCHECK.

R=rjkroege@chromium.org
Bug:578890
Change-Id: Ic2b6fc5b9853a1aec7e5e70a0e979dc6c9e87aea
---
 ui/ozone/platform/wayland/wayland_screen.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ui/ozone/platform/wayland/wayland_screen.cc b/ui/ozone/platform/wayland/wayland_screen.cc
index a7a0b8b1172e..19fb4050b2c0 100644
--- a/ui/ozone/platform/wayland/wayland_screen.cc
+++ b/ui/ozone/platform/wayland/wayland_screen.cc
@@ -100,7 +100,9 @@ display::Display WaylandScreen::GetPrimaryDisplay() const {
 display::Display WaylandScreen::GetDisplayForAcceleratedWidget(
     gfx::AcceleratedWidget widget) const {
   auto* wayland_window = connection_->GetWindow(widget);
-  DCHECK(wayland_window);
+  // A window might be destroyed by this time on shutting down the browser.
+  if (!wayland_window)
+    return GetPrimaryDisplay();
 
   const std::set<uint32_t> entered_outputs_ids =
       wayland_window->GetEnteredOutputsIds();
-- 
2.17.1

